### ISSUE TITLE
Sample does not run on iOS - Android throws intermittent exceptions

### ISSUE THREAD

**Issue by RocketPop** [Azure-Samples/active-directory-b2c-xamarin-native#64]
Build env: Visual studio 2017 15.8.3

Running this sample "out of the box" on Android (simulator and device) Android API 23-27 often gives an error with invalid Oauth2 reponse that it cannot parse. This happens with both my AD / tenant and the sample connection info "as is".

Running this sample "out of the box" on iOS (simulator and device) iOS 11-11.4 gives an error regarding type lookups and never displays any webview. Also tried setting the `useEmbeddedWebview` where the  UiParent is constructed... Project is in Debug, set to do not link. 

```
"System.TypeLoadException: Could not set up parent class, due to: Could not load type of field 'Microsoft.Identity.Core.UI.WebviewBase:asWebAuthenticationSession' (5) due to: Could not resolve type with token 0100004d (from typeref, class/assembly AuthenticationServices.ASWebAuthenticationSession, Xamarin.iOS, Version=0.0.0.0, Culture=neutral, PublicKeyToken=84e04ff9cfb79065) assembly:Xamarin.iOS, Version=0.0.0.0, Culture=neutral, PublicKeyToken=84e04ff9cfb79065 type:AuthenticationServices.ASWebAuthenticationSession member:(null) signature:<none> assembly:/Users/wilks/Library/Developer/CoreSimulator/Devices/99CBFEFD-157F-49AB-8A7A-FFFA876F5BE0/data/Containers/Bundle/Application/A7F2516F-663C-4CDD-BF23-64BEF59C058F/UserDetailsClientiOS.app/Microsoft.Identity.Client.dll type:WebviewBase member:(null) signature:<none>\n  at Microsoft.Identity.Client.PublicClientApplication.CreateWebAuthenticationDialog (Microsoft.Identity.Client.UIParent parent, Microsoft.Identity.Client.UIBehavior behavior, Microsoft.Identity.Core.RequestContext requestContext) [0x00015] in <f077f6b3700d43e3adf74b7a06b732ac>:0 \n  at Microsoft.Identity.Client.PublicClientApplication+<AcquireTokenForUserCommonAsync>d__23.MoveNext () [0x00088] in <f077f6b3700d43e3adf74b7a06b732ac>:0 \n--- End of stack trace from previous location where exception was thrown ---\n  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x00037] in /Library/Frameworks/Xamarin.iOS.framework/Versions/11.14.0.14/src/Xamarin.iOS/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:187 \n  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in /Library/Frameworks/Xamarin.iOS.framework/Versions/11.14.0.14/src/Xamarin.iOS/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 \n  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in /Library/Frameworks/Xamarin.iOS.framework/Versions/11.14.0.14/src/Xamarin.iOS/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 \n  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/11.14.0.14/src/Xamarin.iOS/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:535 \n  at Microsoft.Identity.Client.PublicClientApplication+<AcquireTokenAsync>d__16.MoveNext () [0x000a6] in <f077f6b3700d43e3adf74b7a06b732ac>:0 \n--- End of stack trace from previous location where exception was thrown ---\n  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x00037] in /Library/Frameworks/Xamarin.iOS.framework/Versions/11.14.0.14/src/Xamarin.iOS/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:187 \n  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in /Library/Frameworks/Xamarin.iOS.framework/Versions/11.14.0.14/src/Xamarin.iOS/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 \n  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in /Library/Frameworks/Xamarin.iOS.framework/Versions/11.14.0.14/src/Xamarin.iOS/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 \n  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/11.14.0.14/src/Xamarin.iOS/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:357 \n  at UserDetailsClient.Core.MainPage+<OnSignInSignOut>d__1.MoveNext () [0x0012d] in D:\\Dev\\Research\\adb2c\\UserDetailsClient\\UserDetailsClient.Core\\MainPage.xaml.cs:26 "
```
Does this sample work on iOS for anyone? I can find NO working samples or code snippets for B2C on iOS with Xamarin forms anywhere. Untested on iOS 10.x

**Comment by RocketPop** [comment]
Ok after looking a bit more, it appears that the instructions here: [to use b2clogin.com ](https://docs.microsoft.com/en-us/azure/active-directory-b2c/b2clogin#update-your-application) don't work (leading to those parse errors), when I made my b2c flows the settings were pointing to b2clogin.com... These worked if I made a big manual HTTP request as shown here: [manual link](https://docs.microsoft.com/en-us/azure/active-directory-b2c/active-directory-b2c-access-tokens#requesting-a-token) or running the flow in the portal. But changing everything back to login.microsoftonline.com works with MSAL and got me past that Oauth2 parse error. Secondly, the custom redirect URI that I set up in the registered application seems to be ignored. Looking in the response, it always returns with a response in the form `$"msal{tenantID}://auth"` regardless of the Custom Redirect URI on the app properties blade. I see your sample also registers the custom uri scheme in this way.  I don't see anywhere that behavior is documented... (I see it's "recommended" on that blade)... So then I added this format to the registered application CustomRedirect URI panel and the login behavior finally worked. So I guess this is mostly a documentation problem instead of a bug - except for b2clogin.com not seeming to work.

**Comment by RocketPop** [comment]
Even more experimentation. I deleted the entire directory and b2c clients and recreated the entire setup - being sure to only use the login.microsoftonline.com access. Now the sample does work on both platforms (the type load exception does not occur anymore either - perhaps I mixed some V2 preview flows?). However now I see that B2C profile editing for local accounts [does not work](https://stackoverflow.com/questions/40942245/pass-token-or-username-to-b2c-edit-policy-in-msal) and has not worked for a year despite a remark on User Voice from a team member in April of 2017 that it is coming to preview "in a few weeks". There are also many cosmetic glitches on iOS with zooming into text fields and buttons...  I think that for native apps in Xamarin - the B2C product is not suitable at this time.
